### DISCUSSION TITLE
Will Gateway API ever be shipped in native Kubernetes ?

### DISCUSSION THREAD

**Discussion by adamency** [kubernetes-sigs/gateway-api#3017] [General]
Will Gateway API ever be shipped in native Kubernetes like Ingress is, i.e. will `gateway.networking.k8s.io` ever join a default api group of the kube-apiserver ?

GA was announced for Gateway API at the end of last year, but not inclusion in Kubernetes. I couldn't find this info anywhere.

**Comment by robscott** [comment]
Great question! I don't think Gateway API will ever be part of kubernetes/kubernetes itself, but there has been growing interest in more and more cluster provisioning tools + cluster providers installing the GA APIs by default in new clusters. So while I think we'll continue to keep our versioning and release disconnected from k/k, I expect the latest GA release of Gateway API will be available by default in a lot more environments over the coming years.

**Reply by adamency** [reply]
Thanks a lot for your asnwer @robscott. My concern was mainly about bare-metal installations, but good to know. Do you know if there is maybe a mechanism set up to create a k8 cluster with Gateway as a hard replacement of Ingress cluster-wide. I'm thinking for an instance of a flag for kubeadm to install them on `kubeadm init` and then an operator that would serve to automatically replace all Ingress resources created by their Gateway equivalent for all manifests and charts pulled out from the web. That looks like this would be possible, provided you already have a gateway controller installed beforehand on your cluster. Are you aware of any community project similar to this ?
